server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/DLPRulesRoutesTest.java

```diff
@@ -0,0 +1,410 @@
+        .put(ID, "5")
+        .put(EXPLANATION, "explanation 4")
+        .put(TARGETS_SENDER, false)
+        .put(TARGETS_RECIPIENTS, true)
+        .put(TARGETS_CONTENT, false);
+
+    private static final DLPConfigurationItem CONFIGURATION_ITEM_1 = DLPConfigurationItem.builder()
+        .id(DLPConfigurationItem.Id.of("1"))
+        .explanation("explanation 1")
+        .expression(DEFAULT_DOMAIN)
+        .targetsSender()
+        .targetsRecipients()
+        .targetsContent()
+        .build();
+    private static final DLPConfigurationItem CONFIGURATION_ITEM_2 = DLPConfigurationItem.builder()
+        .id(DLPConfigurationItem.Id.of("2"))
+        .expression(DEFAULT_DOMAIN)
+        .targetsSender()
+        .build();
+    private static final List<DLPConfigurationItem> CONFIGURATION_ITEMS = ImmutableList.of(CONFIGURATION_ITEM_1, CONFIGURATION_ITEM_2);
+
+    private WebAdminServer webAdminServer;
+    private EventSourcingDLPConfigurationStore dlpStore;
+
+    private void createServer(DLPConfigurationStore dlpConfigurationStore, DomainList domainList) throws Exception {
+        webAdminServer = WebAdminUtils.createWebAdminServer(
+            new DefaultMetricFactory(),
+            new DLPRulesRoutes(dlpConfigurationStore, domainList, new JsonTransformer()));
+        webAdminServer.configure(NO_CONFIGURATION);
+        webAdminServer.await();
+
+        requestSpecification = buildRequestSpecification(webAdminServer);
+    }
+
+    RequestSpecification buildRequestSpecification(WebAdminServer server) {
+        return new RequestSpecBuilder()
+            .setContentType(ContentType.JSON)
+            .setAccept(ContentType.JSON)
+            .setBasePath(DLPRulesRoutes.BASE_PATH)
+            .setPort(server.getPort().getValue())
+            .setConfig(newConfig().encoderConfig(encoderConfig().defaultContentCharset(StandardCharsets.UTF_8)))
+            .build();
+    }
+
+
+    @BeforeEach
+    void setup() throws Exception {
+        DNSService dnsService = mock(DNSService.class);
+        Mockito.when(dnsService.getHostName(any())).thenReturn("localhost");
+        Mockito.when(dnsService.getLocalHost()).thenReturn(InetAddress.getByName("localhost"));
+
+        MemoryDomainList domainList = new MemoryDomainList(dnsService);
+        domainList.setAutoDetectIP(false);
+        domainList.addDomain(SENDER_DOMAIN);
+
+        dlpStore = new EventSourcingDLPConfigurationStore(new InMemoryEventStore());
+        createServer(dlpStore, domainList);
+    }
+
+
+    @Nested
+    class DefineStore {
+        @Test
+        void putShouldstoreTheConfigurations() {
+            String body = JsonNodeFactory.instance.arrayNode()
+                .add(JSON_WITH_POSITIVE_TARGETS)
+                .add(JSON_WITH_NEGATIVE_TARGETS)
+                .toString();
+
+            given()
+                .body(body)
+            .when()
+                .put(DEFAULT_DOMAIN)
+            .then()
+            .log().all()
```

Comment:
That, is a debug statement generating verbose builds output.

Please do use a generic rest assured setUp (configured globally in your SetUp method) .logIfValidationFails